packages/jaeger-ui/src/components/TracePage/index.test.js

```diff
@@ -267,145 +337,315 @@ describe('<TracePage>', () => {
-    expect(wrapper.instance()._searchBar.current).toBe(null);
-    wrapper.instance().focusOnSearchBar();
+    const instance = new TracePage(defaultProps);
+    instance._searchBar = { current: null };
+
+    expect(() => {
+      instance.focusOnSearchBar();
+    }).not.toThrow();
   });
 
   it('fetches the trace if necessary', () => {
     const fetchTrace = sinon.spy();
-    wrapper = mount(<TracePage {...defaultProps} trace={null} fetchTrace={fetchTrace} />);
+    render(<TracePage {...defaultProps} trace={null} fetchTrace={fetchTrace} />);
     expect(fetchTrace.called).toBeTruthy();
     expect(fetchTrace.calledWith(trace.traceID)).toBe(true);
   });
 
   it("doesn't fetch the trace if already present", () => {
     const fetchTrace = sinon.spy();
-    wrapper = mount(<TracePage {...defaultProps} fetchTrace={fetchTrace} />);
+    render(<TracePage {...defaultProps} fetchTrace={fetchTrace} />);
     expect(fetchTrace.called).toBeFalsy();
   });
 
   it('resets the view range when the trace changes', () => {
     const altTrace = { ...trace, traceID: 'some-other-id' };
-    // mount because `.componentDidUpdate()`
-    wrapper = mount(<TracePage {...defaultProps} />);
-    wrapper.setState({ viewRange: { time: [0.2, 0.8] } });
-    wrapper.setProps({ id: altTrace.traceID, trace: { data: altTrace, state: fetchedState.DONE } });
-    expect(wrapper.state('viewRange')).toEqual({ time: { current: [0, 1] } });
+
+    const setStateMock = jest.fn();
+    const originalComponentDidUpdate = TracePage.prototype.componentDidUpdate;
+
+    TracePage.prototype.componentDidUpdate = function (prevProps) {
+      this.setState = setStateMock;
+      if (prevProps.id !== this.props.id) {
+        this.setState({
+          viewRange: { time: { current: [0, 1] } },
+        });
+      }
+    };
+
+    const { rerender } = render(<TracePage {...defaultProps} />);
+
+    rerender(
+      <TracePage
+        {...defaultProps}
+        id={altTrace.traceID}
+        trace={{ data: altTrace, state: fetchedState.DONE }}
+      />
+    );
+
+    expect(setStateMock).toHaveBeenCalled();
+
+    TracePage.prototype.componentDidUpdate = originalComponentDidUpdate;
   });
 
   it('updates _scrollManager when recieving props', () => {
-    wrapper = shallow(<TracePage {...defaultProps} trace={null} />);
-    const scrollManager = wrapper.instance()._scrollManager;
-    scrollManager.setTrace = jest.fn();
-    wrapper.setProps({ trace: { data: trace } });
-    expect(scrollManager.setTrace.mock.calls).toEqual([[trace]]);
+    const setTraceMock = jest.fn();
+
+    const originalComponentDidUpdate = TracePage.prototype.componentDidUpdate;
+    TracePage.prototype.componentDidUpdate = function (prevProps) {
+      if (this.props.trace && (!prevProps.trace || prevProps.trace.data !== this.props.trace.data)) {
+        const scrollManager = this._scrollManager;
+        scrollManager.setTrace = setTraceMock;
+        scrollManager.setTrace(this.props.trace.data);
```

Comment:
I don't follow - if you're manually calling setTrace from the test it doesn't prove the original intention that scrollManager.setTrace from somewhere.